src/Carousel.tsx

```diff
@@ -126,30 +126,33 @@ class Carousel extends React.Component<CarouselProps, CarouselInternalState> {
   public setClones(
     slidesToShow: number,
     itemWidth?: number,
-    forResizing?: boolean
+    forResizing?: boolean,
+    resetCurrentSlide: boolean = false
   ): void {
     // if forResizing is true, means we are on client-side.
     // if forResizing is false, means we are on server-side.
     // because the first time we set the clones, we change the position of all carousel items when entering client-side from server-side.
     // but still, we want to maintain the same position as it was on the server-side which is translateX(0) by getting the couter part of the original first slide.
     this.isAnimationAllowed = false;
     const childrenArr = React.Children.toArray(this.props.children);
-    const initialSlide = getInitialSlideInInifteMode(
+    const initialSlide = getInitialSlideInInfiniteMode(
       slidesToShow || this.state.slidesToShow,
       childrenArr
     );
     const clones = getClones(this.state.slidesToShow, childrenArr);
-    if (!notEnoughChildren(this.state, this.props, slidesToShow)) {
-      this.setState(
-        {
-          totalItems: clones.length,
-          currentSlide: forResizing ? this.state.currentSlide : initialSlide
-        },
-        () => {
-          this.correctItemsPosition(itemWidth || this.state.itemWidth);
-        }
-      );
-    }
+    const currentSlide =
+      childrenArr.length < this.state.slidesToShow
+        ? 0
+        : this.state.currentSlide;
+    this.setState(
+      {
+        totalItems: clones.length,
+        currentSlide: forResizing && !resetCurrentSlide ? currentSlide : initialSlide,
+      },
+      () => {
+        this.correctItemsPosition(itemWidth || this.state.itemWidth);
```

Comment:
The changes in this function is not needed, as it shouldn't be called at all. ``` TotalItems``` will be default to ```this.props.children.length``` if not specified.